src/reports/reports.service.ts

```
@@ -643,6 +688,130 @@ export class ReportsService {
+      if (!fieldNameToFieldMap.has(fieldName)) {
+        throw new BadRequestException(
+          `Field with name '${fieldName}' not found in report`,
+        );
+      }
+    }
+
+    const existingDataByFieldName = new Map(
+      submission.submissionData.map((sd) => [sd.reportField.name, sd]),
+    );
+
+    await this.connection.transaction(async (manager) => {
```

ESLint

[error] Unexpected space before function parentheses.

(space-before-function-paren)